ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala

```diff
@@ -747,6 +748,32 @@ abstract class AbstractWebsocketServiceIntegrationTest
     go(createCount)
   }
 
+  /** Updates the ACS retrieved with [[readAcsN]] with the given number of events
+    * The caller is in charge of reading the live marker if that is expected
+    */
+  private[this] def updateAcs(
+      acs: Vector[(String, JsValue)],
+      events: Int,
+  ): Consume.FCC[JsValue, Map[String, JsValue]] = {
+    val dslSyntax = Consume.syntax[JsValue]
+    import dslSyntax._
+    def go(
+        acs: Map[String, JsValue],
+        missingEvents: Int,
+    ): Consume.FCC[JsValue, Map[String, JsValue]] =
+      if (missingEvents <= 0) {
+        point(acs)
+      } else {
+        for {
+          ContractDelta(creates, deletes, _) <- readOne
+          newAcs = acs ++ creates -- deletes.map(_.contractId.unwrap)
+          events = creates.size - deletes.size
```

Comment:
> It’s not the total number of events since deletes (which maybe should be called archives) are clearly also events.

The intent was actually this, I simply put a `-` there but I meant `+`. It works simply because I only have creates in tests.

> which maybe should be called archives

In principle I agree, I assumed that the type parameter `D` in the various domain objects representing events in the JSON API was meant to mean `Delete` and wanted to go with that, but thinking better about it could have been picked because it came after `C` in the alphabet (which I assume stands for `Create`). Changing this,

Reply:
Addressed by https://github.com/digital-asset/daml/pull/9847/commits/178fd6ccee57b84e434cbc5df72edc5ee1f07407